khmer/_khmermodule.cc

```diff
@@ -4201,6 +4202,221 @@ static void khmer_subset_dealloc(PyObject* self)
+static int khmer_hllcounter_init(khmer_KHLLCounterObject * self,
+                                 PyObject * args, PyObject * kwds);
+static PyObject * khmer_hllcounter_getattr(PyObject * obj, char * name);
+
+static PyTypeObject khmer_KHLLCounterType = {
+    PyObject_HEAD_INIT(NULL)
+    0,
+    "KHLLCounter", sizeof(khmer_KHLLCounterObject),
+    0,
+    (destructor)khmer_hllcounter_dealloc,	/*tp_dealloc*/
+    0,				/*tp_print*/
+    khmer_hllcounter_getattr,	/*tp_getattr*/
+    0,				/*tp_setattr*/
+    0,				/*tp_compare*/
+    0,				/*tp_repr*/
+    0,				/*tp_as_number*/
+    0,				/*tp_as_sequence*/
+    0,				/*tp_as_mapping*/
+    0,				/*tp_hash */
+    0,				/*tp_call*/
+    0,				/*tp_str*/
+    0,				/*tp_getattro*/
+    0,				/*tp_setattro*/
+    0,				/*tp_as_buffer*/
+    Py_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE,		/*tp_flags*/
+    "HyperLogLog counter object",           /* tp_doc */
+    0,                       /* tp_traverse */
+    0,                       /* tp_clear */
+    0,                       /* tp_richcompare */
+    0,                       /* tp_weaklistoffset */
+    0,                       /* tp_iter */
+    0,                       /* tp_iternext */
+    0,  /* tp_methods */
+    0,                       /* tp_members */
+    0,                       /* tp_getset */
+    0,                       /* tp_base */
+    0,                       /* tp_dict */
+    0,                       /* tp_descr_get */
+    0,                       /* tp_descr_set */
+    0,                       /* tp_dictoffset */
+    (initproc)khmer_hllcounter_init,   /* tp_init */
+    0,                       /* tp_alloc */
+};
+
+#define is_hllcounter_obj(v)  ((v)->ob_type == &khmer_KHLLCounterType)
+
+static PyObject* khmer_hllcounter_new(PyTypeObject * type, PyObject * args,
+                                      PyObject * kwds)
+{
+    khmer_KHLLCounterObject * self;
+    self = (khmer_KHLLCounterObject *)type->tp_alloc(type, 0);
+
+    if (self != NULL) {
+        double error_rate = 0;
+        WordLength ksize = 0;
+
+        if (!PyArg_ParseTuple(args, "db", &error_rate, &ksize)) {
+            Py_DECREF(self);
+            return NULL;
+        }
+
+        if ((error_rate < 0) || (error_rate > 1.0)) {
+            Py_DECREF(self);
+            PyErr_SetString(PyExc_ValueError,
+                            "Error rate should be between 0.0 and 1.0");
+            return NULL;
+        }
+
+        try {
+            self->hllcounter = new HLLCounter(error_rate, ksize);
+        } catch (khmer_exception &e) {
+            PyErr_SetString(PyExc_ValueError, e.what());
+            return NULL;
+        }
+    }
+
+    return (PyObject *) self;
+}
+
+static int khmer_hllcounter_init(khmer_KHLLCounterObject * self,
+                                 PyObject * args, PyObject * kwds)
+{
+    return 0;
+}
+
+//
+// khmer_hllcounter_dealloc -- clean up a hllcounter object.
+//
+
+static void khmer_hllcounter_dealloc(PyObject* obj)
```

Comment:
The `obj` argument type can be `khmer_KHLLCounterObject` directly; no need to cast below.